src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -156,7 +157,7 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
     if (this.state.showError) {
       return <DataSourceErrorMenu />;
     }
-    const label = this.state.selectedOption.length > 0 ? this.state.selectedOption[0].label : '';
+    const label = this.state.selectedOption.length > 0 ? this.state.selectedOption[0].label! : '';
```

Comment:
Wouldn't `label` be undefined here at initial render of the component if consumers pass in `[{id}]`?

Reply:
Good catch! Let me add check here